classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php

```diff
@@ -92,15 +92,24 @@ function fetch($request) {
 			}
 			$templateMgr->assign('authorUserGroupOptions', $userGroupNames);
 		} else {
-			// The user doesn't have any author user group assignments.  They should be either a manager.
+			// The user doesn't have any author user group assignments.
 			// Add all manager user groups
 			$managerUserGroupAssignments = $userGroupAssignmentDao->getByUserId($user->getId(), $this->context->getId(), ROLE_ID_MANAGER);
-			if($managerUserGroupAssignments) while($managerUserGroupAssignment = $managerUserGroupAssignments->next()) {
-				$managerUserGroup = $userGroupDao->getById($managerUserGroupAssignment->getUserGroupId());
-				$userGroupNames[$managerUserGroup->getId()] = $managerUserGroup->getLocalizedName();
+			if(!$managerUserGroupAssignments->wasEmpty()){ 
+				while($managerUserGroupAssignment = $managerUserGroupAssignments->next()) {
+					$managerUserGroup = $userGroupDao->getById($managerUserGroupAssignment->getUserGroupId());
+					$userGroupNames[$managerUserGroup->getId()] = $managerUserGroup->getLocalizedName();
+				}
+				$templateMgr->assign('authorUserGroupOptions', $userGroupNames);
```

Comment:
That is part of the original code (I just fixed a small thing there). 

I guess the idea is that most editors do not actually have an author role but tend to submit things like editorials, maybe?

Reply:
Yeah I'm trying to think if it makes a difference. But will `authorUserGroupOptions` result in a selection control in the submission step? If so, might it be confusing to be asked in what role they're submitting and not have access to authorial roles?

Reply:
_If_ the author roles are carried through in the submission data, then let's make them choose a role.

For example, if a manager selects Translator, will they be assigned that role in the Participants list? If so, then it's good data to collect. Otherwise, just let them get on with the submission without choosing a role.

Reply:
At the moment in master branch:
- if editor submits a manuscript and only has manager roles, the editor will be added to the contributors list of that submission, but not as an author in the stage assignments. This is probably what most editors expect when they send things like editorials.

- If we would add the author role selection also for the users with only manager roles, then that role would be carried on to the stage assignments. But, I think that some editors still want to submit papers as editors

So maybe:
Case 1: users without any roles: show selection of author roles to choose from
Case 2: manager users with author roles: show selection of both manager roles and author roles and let the editor choose 
Case 3: manager users without author roles: show selection of both manager roles and author roles and let editor choose. If she chooses an author role, enroll the user as an author

I am thinking that this would solve the situations where editors want to submit to their own journal and receive a peer-review for their paper. In these situations they of course choose an author role. In other cases, like when sending in editorials etc. they can choose a managerial role. But I am not sure if this is apparent for the editors without some guidance?

Reply:
I agree with all of your cases.

> I am thinking that this would solve the situations where editors want to submit to their own journal and receive a peer-review for their paper.

Yes, but we still can't guarantee blind review, since a `ROLE_ID_MANAGER` can still use the login-as function to access anything.

Reply:
Sure and the secondary author problem also still applies. But having that selection with perhaps a few guiding words would make things more clear for editors and they do not have to guess what will happen when they submit something.